test/test_unistd.rs

```diff
@@ -156,6 +150,15 @@ macro_rules! execve_test_factory(
     )
 );
 
+#[cfg(not(target_os = "android"))]
```

Comment:
This will be cleaner if you use `cfg_if!` and group the platform-specific code together.